src/Symfony/Component/HttpKernel/Event/ControllerEvent.php

```diff
@@ -79,29 +79,33 @@ public function setController(callable $controller, array $attributes = null): v
     }
 
     /**
-     * @return array<class-string, list<object>>
+     * @template T of class-string|null
+     *
+     * @param T $className
+     *
+     * @return array<class-string, list<object>>|list<object>
+     *
+     * @psalm-return (T is null ? array<class-string, list<object>> : list<object>)
      */
-    public function getAttributes(): array
+    public function getAttributes(string $className = null): array
     {
-        if (isset($this->attributes)) {
-            return $this->attributes;
```

Comment:
I would prefer keeping the early return
`return null === $className ? $this->attributes : $this->attributes[$className] ?? [];`